src/types/local_date.rs

```diff
@@ -0,0 +1,271 @@
+    pub year: Option<i32>,
+    pub month: Option<i8>,
+    pub day: Option<i8>,
+    pub in_date: bool, // value can be represented as Date class in JS
+}
+
+#[napi]
+impl LocalDateWrapper {
+    /// Create a new object from the day, month and year.
+    #[napi]
+    pub fn new(day: i8, month: i8, year: i32) -> napi::Result<Self> {
+        Self::check_valid_date(day, month, year)
+    }
+
+    /// Create a new object from number of days since 01.01.1970.
+    #[napi]
+    pub fn new_day(value: i32) -> napi::Result<Self> {
+        if !(MIN_JS_DATE..=MAX_JS_DATE).contains(&value) {
+            Ok(LocalDateWrapper {
+                value,
+                year: None,
+                month: None,
+                day: None,
+                in_date: false,
+            })
+        } else {
+            let date = Self::date_from_days(value.into());
+            Ok(LocalDateWrapper::new(date.0, date.1, date.2)?)
+        }
+    }
+
+    /// Counts the number of days since 01.01.1970.
+    fn days_since_epoch(day: i8, month: i8, year: i32) -> i32 {
+        let mut total_days = 0;
+
+        let number_day = DAY_IN_MONTH[month as usize - 1] // number of days from 1 January
+            + if Self::is_leap_year(year) && month > 2 {
+                1
+            } else {
+                0
+            }
+            + day as i32
+            - 1;
+
+        if year >= 1970 {
+            total_days += (year - 1970) * 365 + Self::number_leap_years(year - 1) - LEAP_YEAR_1970
+                + number_day;
+        } else {
+            total_days -= if Self::is_leap_year(year) { 366 } else { 365 } - number_day;
+            if year < 0 {
+                total_days -= (1970 - year - 1) * 365
+                    + Self::number_leap_years(year.abs() + 1)
+                    + LEAP_YEAR_1970
+                    + 1;
+            } else {
+                total_days -=
+                    (1970 - year - 1) * 365 + LEAP_YEAR_1970 - Self::number_leap_years(year + 1);
+            }
+        }
+        total_days
+    }
+
+    /// Adds the number of days to self.
+    fn date_from_days(mut n: i64) -> (i8, i8, i32) {
+        let mut year = 1970;
+        while n.abs() >= 365 {
+            // Find the number of years in n days.
+            let k = (n / 365) as i32;
+            if year > 0 && year + k < 0 {
+                n += 1;
+            }
+            year += k;
+
+            if k > 0 {
+                n -= (k.abs() * 365 + Self::number_leap_years(year - 1)
+                    - Self::number_leap_years(year - k)) as i64;
+            } else {
+                n += (k.abs() * 365 + Self::number_leap_years(year - k)
+                    - Self::number_leap_years(year + 1)) as i64;
+            }
+        }
+
+        if n < 0 {
+            // If the remaining number of days is negative, change the year and count the complement.
+            year -= 1;
+            n += if Self::is_leap_year(year) { 366 } else { 365 };
+        }
```

Comment:
Can you guarantee we will always be at most one year behind?

Reply:
It seems like this may not be the case for dates ~1500 years in the future

Reply:
If there were `n > 365` then it would not come out of the loop above with the condition: 
`while n.abs() >= 365`